flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartitionView.java

```diff
@@ -50,12 +50,14 @@
 
 	boolean isReleased();
 
+	void resumeConsumption();
+
 	Throwable getFailureCause();
 
 	/**
-	 * Returns whether the next buffer is an event or not.
+	 * Returns true if we can read data without available credits.
 	 */
-	boolean nextBufferIsEvent();
+	boolean isAvailableWithoutCredit();
```

Comment:
it is better to merge `isAvailableWithoutCredit()` with existing `isAvailable`() to avoid maintaining two separate interface methods. We can integrate them into `isAvailable(int credit)` method for distinguishing the condition inside implementations.